bot/exts/evergreen/conversationstarters.py

```diff
@@ -24,6 +23,10 @@
     # All the allowed channels that the ".topic" command is allowed to be executed in.
     ALL_ALLOWED_CHANNELS = [channel_id for channel_id in PY_TOPICS.keys()] + list(WHITELISTED_CHANNELS)
 
+# Putting all topics into one dictionary and shuffling lists to reduce same-topic repetitions.
+all_topics = {'default': STARTERS, **PY_TOPICS}
+TOPICS = {channel: RandomCycle(topics) if len(topics) else False for channel, topics in all_topics.items()}
```

Comment:
This can be simplified to
```suggestion
TOPICS = {
    channel: RandomCycle(topics or ['No topics found for this channel.'])
    for channel, topics in all_topics.items()
}
```
And then you can remove a layer of checking when getting topics in line 60 where we do not have to check if `TOPICS[ctx.channel.id]` is `False` or not. like so
```py
        try:
            # Fetching topics.
            channel_topics = TOPICS[ctx.channel.id]

        # If the channel isn't Python-related.
        except KeyError:
            embed.title = f'**{next(TOPICS["default"])}**'

        # If the channel ID doesn't have any topics.
        else:
            embed.title = f'**{next(channel_topics)}**'
```

Reply:
I like these simplifications. Thanks for the input!